container.go

```diff
@@ -72,11 +74,15 @@ func (noopWriteCloser) Close() error {
 
 // WithTTY will run the container with the provided in/out/err connected to the
 // tty in the container.  Resize events will automatically be propagated.
-func WithTTY(in FdReader, out, err io.Writer) ContainerOption {
+func WithTTY(in FdReader, outW, errW io.Writer) ContainerOption {
 	return containerOptionFunc(func(co *ContainerOptions) {
 		co.Tty = true
-		WithInput(in).SetContainerOptions(co)
-		WithOutput(out, err).SetContainerOptions(co)
+		inReader, err := cancelreader.NewReader(in)
+		if err != nil {
+			panic(fmt.Sprintf("create cancel reader: %s", err))
+		}
+		WithInput(inReader).SetContainerOptions(co)
```

Comment:
overall this looks good, but the panic is pretty icky.  Maybe we update ContainerOption to make  SetContainerOptions return an error?   Also we should probably move this logic to `WithInput`, I don't think TTY is the only case we would want to cancel a read on the input?

Reply:
Not sure about moving this to `WithInput`. The cancelable reader only works on files, so it would have to be best-effort, which I don't really like.

I can make this return an error if you want. I don't expect the panic to happen in practice for the TTY case (and if it does, it's pretty much unrecoverable anyway).

Reply:
I read through more of the NewReader, I agree the best-effort non-file reader seems problematic.  The error conditions seem like failures to create NewPipe or init Epoll both of which are likely fatal anyway, so seems good to me.